src/AWS.Deploy.Orchestration/DeploymentSettingsHandler.cs

```diff
@@ -25,17 +26,25 @@ public interface IDeploymentSettingsHandler
         /// Iterates over the option setting values found at <see cref="DeploymentSettings"/> and applies them to the selected recommendation
         /// </summary>
         Task ApplySettings(DeploymentSettings deploymentSettings, Recommendation recommendation, IDeployToolValidationContext deployToolValidationContext);
+
+        /// <summary>
+        /// Save the deployment settings at the specified file path
+        /// </summary>
+        /// <returns>An error message if the operation is not successful. The error message will be empty if it is successful</returns>
```

Comment:
It doesn't return an error message, an exception is thrown if is not successful.

Reply:
Fixed this to fail the deployment if the operation to save the settings file fails.